app/controllers/api/v0/pages_controller.rb

```diff
@@ -26,13 +26,41 @@ def index
       if !should_allow_versions && should_include_versions
         raise Api::NotImplementedError, 'The ?include_versions query argument has been disabled.'
       elsif should_allow_versions && should_include_versions
-        results = query
-          .where(uuid: page_ids)
-          .includes(:versions)
-          .where(versions: { different: true })
-          .order('versions.capture_time DESC')
-        lightweight_query(results, &method(:format_page_json))
-      elsif should_include_latest || should_include_earliest
+        if use_lightweight_query
+          results = query
+            .where(uuid: page_ids)
+            .includes(:versions)
+            .where(versions: { different: true })
+            .order('versions.capture_time DESC')
+          lightweight_query(results, &method(:format_page_json))
+        else
+          oj_mode = :rails
+          # Including versions, tags, and maintainers is perfectly OK, but
+          # when you add the WHERE clause for versions, ActiveRecord issues
+          # one extra query per page for the tags and maintainers. Instead,
+          # issue two separate queries and manually plug them together.
+          results = Page
+            .where(uuid: page_ids)
+            .includes(:versions)
+            .where(versions: { different: true })
+            .order(sorting_params.present? ? sorting_params : 'pages.updated_at DESC')
+            .order('versions.capture_time DESC')
+            .as_json(include: :versions)
+          # Tags and Maintainers
+          additions = Page
+            .where(uuid: page_ids)
+            .includes(:tags, :maintainers)
+            .order(sorting_params.present? ? sorting_params : 'pages.updated_at DESC')
+          # Join them up!
+          results.each_with_index do |page, index|
+            other_page = additions[index]
```

Comment:
[Suggestion] The combination of the two queries is the only thing that makes me (slightly) pause because I can think of vague situations where a numerical index could get out of sync between the two queries. I think it might be slightly safer, and not much less performant, to use a lookup hash keyed on the uuid using  `index_by`:


```ruby
additions  = Page.where...index_by(&:uuid)

results.each do |page|
  other_page = additions[page.uuid]
  ...
end
```

Reply:
Ooh, nice, I did not know about `index_by`!